src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -300,6 +300,15 @@ var ReactCompositeComponent = {
           'componentWillRecieveProps(). Did you mean componentWillReceiveProps()?',
         this.getName() || 'A component',
       );
+
+      if (isPureComponent(Component) && typeof inst.shouldComponentUpdate !== 'undefined') {
+        warning(
+          false,
```

Comment:
Make sure it only warns once so we don't spam the user with multiple warnings, and add a test for this behavior.